desktop/libs/notebook/src/notebook/connectors/sql_alchemy.py

```diff
@@ -251,9 +252,25 @@ def execute(self, notebook, snippet):
         }
       )
 
-    result = connection.execute(statement)
+    result = connection.execute(statement, async_=True)
+
+    logs = []
+
+    if result.cursor and hasattr(result.cursor, 'poll'):
+      status = result.cursor.poll().operationState
+
+      if result.cursor and hasattr(result.cursor, 'fetch_logs'):
+        temp_logs = result.cursor.fetch_logs()
+        for message in temp_logs:
+          logs.append(message)
+
+        while status in (TOperationState.INITIALIZED_STATE, TOperationState.RUNNING_STATE):
+          temp_logs = result.cursor.fetch_logs()
+          for message in temp_logs:
+            logs.append(message)
+
+          status = cursor.poll().operationState
```

Comment:
How come this does not go in fetch_status(), get_logs()?

i.e. execute statement should return right away, then the UI does calls to these endpoints
(here we are going to block in the execute() method which defeats all the purpose of async exec)

Reply:
Probably because you did not update the sync flag below:

`      'sync': False,` -->       'sync': True if ...,

Reply:
Problem still there

Reply:
If I am getting it right, you are asking why we have one `for loop `outside (i.e. not in `while loop`), so the answer is -> queries are finishing before going in the while loop (i.e. in the status loop) so they are not going in ( I tried it they are not going inside so I wrote above code)

Reply:
Having the for loop defeats the full purpose of async, please tweak the logic in `check_status` so